src/Features/Core/Portable/CodeRefactorings/CodeRefactoringService.cs

```diff
@@ -271,4 +302,27 @@ protected override bool TryGetExtensionsFromReference(AnalyzerReference referenc
             return false;
         }
     }
+
+    private record struct ProviderKey(string Language, TextDocumentKind DocumentKind, string DocumentExtension) : IEquatable<ProviderKey>
+    {
+        public bool Equals(ProviderKey other)
+        {
+            return Language == other.Language &&
+                   DocumentKind == other.DocumentKind &&
+                   StringComparer.OrdinalIgnoreCase.Equals(DocumentExtension, other.DocumentExtension);
```

Comment:
why case insensivity here?  do we generally have different providers with the same extension (modulo casing)?

Reply:
If there are several provides registered for ".xaml", ".Xaml" and ".XAML" then they all should be in the same group and apply to MyPage.xaml, right? In that case we need to compare extensions in a case sensitive way

Reply:
my question would be: why do you have providers registered with all those different extensions?

We don't do this in roslyn, for example afaict.  We don't allow different casing for things like Language for example.  not sure why extensions would be the same.

I expect this to change for files on disk.  I don't expect it to be different for code which is doing registration.

Reply:
We also create ProviderKey from TextDocument which can have path with randomly cased extension, e.g. `MyPage.xaml` vs `MyControl.XAML`.

Reply:
Ok.  Let's doc that (in the code, with an actual doc comment). Explaining the data flow patterns taht lead to this acutally coming from a real file path, and as such why this si important.  thanks for the explanation :)